### ISSUE TITLE
Failed to recover key for crate_hash(7b325c0bcf1c8f7e-78e92a3d885642c8) with hash 7b325c0bcf1c8f7e-78e92a3d885642c8

### ISSUE THREAD

**Issue by zhiburt** [rust-lang/rust-clippy#8078]
### Summary

Hi there,

Was doing `cargo clippy` in https://github.com/zhiburt/ptyprocess

And got this error.

`cargo clean` fixed this

But I guess it was necessary to report?

Thank you.

### Version

```text
rustc 1.55.0 (c8dfcfe04 2021-09-06)
binary: rustc
commit-hash: c8dfcfe046a7680554bf4eb612bad840e7631c4b
commit-date: 2021-09-06
host: x86_64-unknown-linux-gnu
release: 1.55.0
LLVM version: 12.0.1
```

### Error output

<details><summary>Backtrace</summary>
  <p>

```bash
[mzhiburt@fedora ptyprocess]$ RUST_BACKTRACE=1 RUST_BACKTRACE=full cargo clippy
    Checking ptyprocess v0.2.0 (/home/mzhiburt/projects/ptyprocess)
thread 'rustc' panicked at 'Failed to recover key for crate_hash(7b325c0bcf1c8f7e-78e92a3d885642c8) with hash 7b325c0bcf1c8f7e-78e92a3d885642c8', compiler/rustc_query_impl/src/lib.rs:54:1
stack backtrace:
   0:     0x7f61bb4b6b40 - std::backtrace_rs::backtrace::libunwind::trace::ha0ad43e8a952bfe7
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x7f61bb4b6b40 - std::backtrace_rs::backtrace::trace_unsynchronized::h6830419c0c4130dc
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f61bb4b6b40 - std::sys_common::backtrace::_print_fmt::h8f3516631ffa1ef5
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f61bb4b6b40 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::he1640d5f0d93f618
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x7f61bb5246bc - core::fmt::write::h88012e1f01caeebf
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/core/src/fmt/mod.rs:1115:17
   5:     0x7f61bb4a82d5 - std::io::Write::write_fmt::h360fa85b30182555
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/io/mod.rs:1665:15
   6:     0x7f61bb4ba89b - std::sys_common::backtrace::_print::ha1f00492f406a015
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x7f61bb4ba89b - std::sys_common::backtrace::print::hd54561b13feb6af3
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x7f61bb4ba89b - std::panicking::default_hook::{{closure}}::h84fe124cd0864662
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:208:50
   9:     0x7f61bb4ba371 - std::panicking::default_hook::h5a8e74a76ce290a7
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:225:9
  10:     0x559a33b11609 - clippy_driver::ICE_HOOK::{{closure}}::{{closure}}::hbbdd4d2b2fd752c8
  11:     0x7f61bb4bb0c9 - std::panicking::rust_panic_with_hook::h67c812a4fe9d4c91
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:626:17
  12:     0x7f61bb4bab87 - std::panicking::begin_panic_handler::{{closure}}::h33f9c1b96af300d7
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:519:13
  13:     0x7f61bb4b703c - std::sys_common::backtrace::__rust_end_short_backtrace::h51bae64be5921f0e
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys_common/backtrace.rs:141:18
  14:     0x7f61bb4baae9 - rust_begin_unwind
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:515:5
  15:     0x7f61bb483aab - std::panicking::begin_panic_fmt::h7ab3d84aa4ae32a2
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:457:5
  16:     0x7f61bdd6aef2 - rustc_query_impl::query_callbacks::is_no_builtins::try_load_from_on_disk_cache::h03c81958a0d980ef
  17:     0x7f61bd397ab8 - rustc_query_system::dep_graph::graph::DepGraph<K>::exec_cache_promotions::h19f270b5c21e9ab1
  18:     0x7f61bdd59d3d - rustc_middle::ty::context::tls::with_context::{{closure}}::h994293f2d4333743
  19:     0x7f61bc63631a - rustc_middle::dep_graph::<impl rustc_query_system::dep_graph::DepKind for rustc_middle::dep_graph::dep_node::DepKind>::with_deps::h49d9559e8ddb34ff
  20:     0x7f61bc6b5a33 - <rustc_query_impl::on_disk_cache::OnDiskCache as rustc_middle::ty::context::OnDiskCache>::serialize::h9dc72b44523596a0
  21:     0x7f61bcb3c4a0 - rustc_middle::ty::context::TyCtxt::serialize_query_result_cache::he77a3fa67b9fe197
  22:     0x7f61bc7ae5c7 - rustc_session::utils::<impl rustc_session::session::Session>::time::h257a58c1152ce7d6
  23:     0x7f61bc7c2441 - rustc_incremental::persist::save::save_in::h5c60d00b5ed9f03a
  24:     0x7f61bc7aea34 - rustc_session::utils::<impl rustc_session::session::Session>::time::hd5254d832dd035c1
  25:     0x7f61bc7b6a5e - rustc_data_structures::sync::join::h138bda93912c4c39
  26:     0x7f61bddf5096 - rustc_middle::dep_graph::<impl rustc_query_system::dep_graph::DepKind for rustc_middle::dep_graph::dep_node::DepKind>::with_deps::h1e829bc57bb9a0ae
  27:     0x7f61bddfeece - rustc_incremental::persist::save::save_dep_graph::h03e6d0f4a3176ecc
  28:     0x7f61bd78ec9c - rustc_interface::queries::<impl rustc_interface::interface::Compiler>::enter::h2007d30ef68567cb
  29:     0x7f61bd77e44c - rustc_span::with_source_map::hc5adfd8eb2dd5236
  30:     0x7f61bd78fc8a - rustc_interface::interface::create_compiler_and_run::h976ed8e22b4f89b8
  31:     0x7f61bd782e89 - scoped_tls::ScopedKey<T>::set::h4a2887b4d5b96c48
  32:     0x7f61bd77f9aa - std::sys_common::backtrace::__rust_begin_short_backtrace::h220030c0c75c9d89
  33:     0x7f61bd77edf5 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hd62830e2741d6b63
  34:     0x7f61bb4c7657 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h6bff7798948b1075
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/alloc/src/boxed.rs:1572:9
  35:     0x7f61bb4c7657 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hc2d25ac38f6b2342
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/alloc/src/boxed.rs:1572:9
  36:     0x7f61bb4c7657 - std::sys::unix::thread::Thread::new::thread_start::hbba5bc368baac205
                               at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/sys/unix/thread.rs:74:17
  37:     0x7f61bb3d9299 - start_thread
  38:     0x7f61bb1bb353 - __clone
  39:                0x0 - <unknown>

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust-clippy/issues/new

note: Clippy version: clippy 0.1.55 (c8dfcfe 2021-09-06)

query stack during panic:
end of query stack
error: could not compile `ptyprocess`
```

  </p>
</details>

**Comment by giraffate** [comment]
I can't reproduce it with `rustc 1.57.0 (f1edd0429 2021-11-29)`. Can you try the current stable?

Also, it doesn't seem to be a bug of Clippy.

**Comment by zhiburt** [comment]
It may not be.

After running cargo clean. I no longer can reproduce it as well .

But before I was constantly getting it.

**Comment by Jarcho** [comment]
Closing as an rustc incremental issues.
